### PR TITLE
s/expected/required/g outputs.

### PR DESCRIPTION
**Pull request by hjoliver**
Terminology consistency tweak. (As discussed offline)

(Some extant doc PRs might need tweaking in the same way, now or post merge).

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

**Comment by hjoliver** [comment]
One review will do; @MetRonnie or first-comer.

**Comment by hjoliver** [comment]
(Branch updated from master).

File: src/7-to-8/summary.rst [hunk 1]
```diff
@@ -106,17 +106,17 @@ separated and states of both can be viewed in the GUI.
...
    * Major Changes::ref:`728.optional_outputs`
-   * User Guide::ref:`User Guide Expected Outputs`
+   * User Guide::ref:`User Guide Required Outputs`
    * User Guide::ref:`User Guide Optional Outputs`
 
-By default, all Cylc 8 tasks are expected to succeed - i.e., success is
-an :term:`expected output <expected output>`. Otherwise they will be marked
+By default, all Cylc 8 tasks are required to succeed - i.e., success is
+a :term:`required output`. Otherwise they will be marked
 as :term:`incomplete tasks<incomplete task>` requiring user intervention.
```
Comment (MetRonnie) [inline src/7-to-8/summary.rst:120]:
```suggestion
as :term:`incomplete tasks<incomplete task>` needing user intervention.
```

File: src/glossary.rst [hunk 2]
```diff
@@ -1476,25 +1476,25 @@ Glossary
 
       .. seealso::
 
-         * :term:`expected output`
+         * :term:`required output`
          * :ref:`Cylc User Guide <User Guide Optional Outputs>`
 
 
-   expected output
+   required output
```
Comment (MetRonnie) [inline src/glossary.rst:1483]: Maybe keep the old term as a synonym?
```suggestion
   required output
   expected output
```

File: src/user-guide/writing-workflows/scheduling.rst [hunk 3]
```diff
@@ -1520,24 +1520,24 @@ in :ref:`Section External Triggers`.
...
 ----------------
 
 .. versionadded:: 8.0.0
 
 :term:`Task outputs <task output>` in the :term:`graph` are either
-:term:`expected <expected output>` (the default) or  :term:`optional <optional
+:term:`required <required output>` (the default) or  :term:`optional <optional
 output>`.
 
 The scheduler expects all task outputs to be completed at runtime, unless they
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/scheduling.rst:1536]:
```suggestion
The scheduler requires all task outputs to be completed at runtime, unless they
```

Comment (hjoliver) [reply src/user-guide/writing-workflows/scheduling.rst:1536]: In this case I was thinking more of "the scheduler expects *required outputs* to be completed".  But fair enough.

**Comment by MetRonnie** [comment]
1 more
https://github.com/cylc/cylc-doc/blob/3d3a63da1be9f6276d430171cb6b139f6618e0ca/src/7-to-8/major-changes/suicide-triggers.rst#L107

**Comment by hjoliver** [comment]
> 1 more
> 
> https://github.com/cylc/cylc-doc/blob/3d3a63da1be9f6276d430171cb6b139f6618e0ca/src/7-to-8/major-changes/suicide-triggers.rst#L107

Damn it, good spotting.  Fixed.

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**